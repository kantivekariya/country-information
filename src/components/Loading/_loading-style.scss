@import "../../common-style";

:root {
  // Custom properties goes here
  --accent: rgb(234, 235, 241);
  --size: 2.5rem;
  --duration: 1s;
}

.spinner {
  position: absolute;
  margin: auto;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  width: var(--size);
  aspect-ratio: 1;
  display: grid;
  mask: conic-gradient(from 22deg, #0001, #000);
  animation: spin var(--duration) steps(8) infinite;

  &:before {
    content: "";
    transform: rotate(45deg);
  }
}

.light-loader {
  &,
  &:before {
    --gradient: linear-gradient(black 0 0);
    background: var(--gradient) 50%/34% 8% space no-repeat,
      var(--gradient) 50%/8% 34% no-repeat space;
  }
  
}

.dark-loader {
  &,
  &:before {
    --gradient: linear-gradient(#fff 0 0);
    background: var(--gradient) 50%/34% 8% space no-repeat,
      var(--gradient) 50%/8% 34% no-repeat space;
  }
 
}

@keyframes spin {
  to {
    transform: rotate(1turn);
  }
}
