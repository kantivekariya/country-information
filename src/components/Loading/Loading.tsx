import "./_loading-style.scss";

interface IProps {
  theme: string;
}

const Loading = ({ theme }: IProps) => (
  <div className="" data-testid="loadingTestId">
    <div className={`spinner ${theme}-loader`}></div>
  </div>
);
export default Loading;
