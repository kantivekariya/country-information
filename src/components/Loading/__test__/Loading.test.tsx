import { cleanup, render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import "@testing-library/jest-dom";
import Loading from "../Loading";

afterEach(cleanup);

const loaderProps = {
  theme: "light",
};

describe("Card List", () => {
  it("renders card list", () => {
    render(<Loading {...loaderProps} />, {
      wrapper: MemoryRouter,
    });
    const cardListElement = screen.getByTestId("loadingTestId");
    expect(cardListElement).toBeInTheDocument();
    expect(cardListElement).not.toBeChecked();
  });

  it("should be matches snapshot", () => {
    const { baseElement } = render(<Loading {...loaderProps} />, {
      wrapper: MemoryRouter,
    });
    expect(baseElement).toMatchSnapshot();
  });
});
