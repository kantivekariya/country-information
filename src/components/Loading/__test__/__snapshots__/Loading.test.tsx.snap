// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card List should be matches snapshot 1`] = `
<body>
  <div>
    <div
      class=""
      data-testid="loadingTestId"
    >
      <div
        class="spinner light-loader"
      />
    </div>
  </div>
</body>
`;
